marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -482,8 +482,23 @@ function handleDuplicateSources (propToValidate, arrayWithDuplicates) {
   return deDupedArray;
 }
 
+function addDocumentMetadataToSearchResults(searchResponse) {
+  searchResponse.results.forEach(result => {
+    let hubMetadata = {};
+    let documentMetadata = xdmp.documentGetMetadata(result.uri);
+    if(documentMetadata) {
+      hubMetadata["lastProcessedByFlow"] = documentMetadata.datahubCreatedInFlow;
+      hubMetadata["lastProcessedByStep"] = documentMetadata.datahubCreatedByStep;
+      hubMetadata["lastProcessedDateTime"] = documentMetadata.datahubCreatedOn;
+      hubMetadata["sources"] = getEntitySources(result.uri);
+    }
+    result["hubMetadata"] = hubMetadata;
+  });
+}
+
 module.exports = {
   addPropertiesToSearchResponse,
   buildPropertyMetadata: buildPropertyMetadata,
-  getEntityInstance: getEntityInstance
+  getEntityInstance: getEntityInstance,
+  addDocumentMetadataToSearchResults
```

Comment:
Can you put this in alphabetical order with the other functions? Makes life easier for future developers to find functions.